### PR TITLE
feat: adds codecov

### LINKED ISSUE #6108: Implement code coverage tooling
As a follow up to #6107, we investigate or start setup of [Codecov](https://about.codecov.io/) in our own library.

### PR DESCRIPTION
**Pull request by AlexanderMelox**
Closes #6108 

_Similar to [this PR opened](https://github.com/carbon-design-system/carbon/pull/17304) in Carbon core_

This PR adds [Codecov](https://about.codecov.io/). Codecov ensures that we comply with test coverage standards and PR reviewers are aware of any coverage changes.

After this PR is merged we will begin seeing Codecov comments on PR's.

[Online tooling for Codecov](https://app.codecov.io/gh/carbon-design-system/ibm-products/tree/AlexanderMelox%3A6108--codecov-tooling)

Screenshot 2024-10-16 at 2 07 46 PM

**Note:** Codecov is set to coverage `target: auto` so that code coverage does not fail PR's until we are ready to enforce 80% coverage

#### What did you change?
```
+ .github/codecov.yml
+ .github/workflows/ci.yml
```

**Changed**

-  Test script now runs with `--collectCoverage` to push the report to Codecov

#### How did you test and verify your work?

- No way to check until opening this PR
- Completed Codecov tutorial on personal repo to make sure I understand how it works. 
- https://github.com/AlexanderMelox/codecov-demo/pull/8

**Next steps**
- [ ] Create an issue on increasing Codecov's target coverage percentage

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -110,7 +114,8 @@ jobs:
         with:
           path: '**/node_modules'
           key:
-            test-community-modules-${{ runner.os }}-${{ hashFiles('yarn.lock') }}
+            test-community-modules-${{ runner.os }}-${{ hashFiles('yarn.lock')
+            }}
```
Comment (elycheea) [inline .github/workflows/ci.yml:118]: We can probably remove this since David has a followup PR to remove the community package.

Comment (AlexanderMelox) [reply .github/workflows/ci.yml]: removed it as to align with David's PR
https://github.com/carbon-design-system/ibm-products/pull/6241/files#

**Comment by AlexanderMelox** [comment]
This PR also removes the `test-community` ci action to align with @davidmenendez PR to avoid conflicts
https://github.com/carbon-design-system/ibm-products/pull/6241/files#

**Review by matthewgallo — APPROVED** [review]

**Review by elycheea — APPROVED** [review]
🚢

**Pull request merged**